transcom/mymove
# MB-15882 add endpoint for creating an approved SITAddressUpdate

## [Jira ticket](https://dp3.atlassian.net/browse/MB-15882)

## Summary

This ticket creates a service object to create approved `SitAddressUpdates` and adds an office endpoint to create a`SitAddressUpdate`. The `mto_service_item_updater.go` service object was also changed to allow `SITDestinationFinalAddress` to be changed. Previously, the `ID` was kept the same and the address record was updated. Now we create a new address record to replace `SITDestinationFinalAddressID` and `SITDestinationFinalAddress`.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

Find a `mto_service_item_id` in your DB where `sit_destination_final_address_id` is not null.
In postman, create a POST request to http://officelocal:3000/service-items/:mtoServiceItemID/sit-address-update/ with this body

```
{
  "newAddress": {
    "streetAddress1": "7 Q St",
    "city": "San Antonio",
    "state": "TX",
    "postalCode": "78227",
  },
  "officeRemarks": "test remarks"
}
```

The endpoint returns the service item. Look at `sitAddressUpdates` for the newly created `sitAddressUpdate`
1. Make sure the `sitAddressUpdate` looks correct.
2. Make sure the `sitDestinationFinalAddress` matches the `newAddress` 


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
